sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1540,7 +1540,8 @@ object EliminateSorts extends Rule[LogicalPlan] {
    */
   private def recursiveRemoveSort(
       plan: LogicalPlan,
-      canRemoveGlobalSort: Boolean): LogicalPlan = {
+      canRemoveGlobalSort: Boolean,
+      canEliminateThroughLocalLimit: Boolean = false): LogicalPlan = {
```

Comment:
do we still need this parameter? where we set it to false?

Reply:
https://github.com/apache/spark/blob/2b1637fde5f6c2203d798e253a475d56a1340257/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L1529
and
https://github.com/apache/spark/blob/2b1637fde5f6c2203d798e253a475d56a1340257/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L1549-L1550

Reply:
do they need to skip LocalLimit?

Reply:
Remove that parameter because the first sort can also be removed:
```
Sort [b#0 ASC NULLS FIRST], true         
+- LocalLimit 2                          
   +- Sort [a#0 ASC NULLS FIRST], true    <------------------- Can be removed
      +- Project [a#0, b#0]              
         +- LocalRelation [a#0, b#0, c#0]
```